core/api/test/integration/services/lnurl-pay.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { decodeInvoice } from "@/domain/bitcoin/lightning"
+import { ErrorFetchingLnurlInvoice } from "@/domain/bitcoin/lnurl/errors"
+import { LnurlPayService } from "@/services/lnurl-pay"
+
+describe("LnurlPayService", () => {
+  const lnurlPayService = LnurlPayService()
+
+  describe("fetchInvoiceFromLnAddressOrLnurl", () => {
+    it("fetches an invoice from an lnurl", async () => {
+      const invoice = await lnurlPayService.fetchInvoiceFromLnAddressOrLnurl({
+        amount: {
+          amount: BigInt(1000),
+          currency: "BTC",
+        } as BtcPaymentAmount,
+        lnAddressOrLnurl:
+          "lnurl1dp68gurn8ghj7urp0yhxymrfde4juumk9uh8wetvdskkkmn0wahz7mrww4excup0w3jhxaqlq07tq", // https://pay.blink.sv/.well-known/lnurlp/test
```

Comment:
please dont implement tests that depends on external resoures

Reply:
This service inherently depends on external resources. How would you propose testing it otherwise? I suppose we could try to mock the response of the http requests it will make. Otherwise bringing up the entire galoy pay and galoy api locally seems like it would introduce a lot of overhead and I dont believe the library's validation supports an ln address of the form `name@localhost:3002` where a port is included.